src/main/webapp/app/exercises/programming/manage/instructions-editor/analysis/programming-exercise-instruction-analysis.component.html

```diff
@@ -29,4 +29,23 @@
             </div>
         </div>
     </ng-template>
+    <ng-container *ngIf="numOfTasks > 15; else tasksOk">
+        <div id="instruction_analysis_num-of-tasks-issues" class="editable-instruction-container__status__item">
+            <fa-icon [icon]="faExclamationTriangle" class="text-warning me-1" [ngbTooltip]="tasksTooltipContent"></fa-icon>
+            <span class="text-info" jhiTranslate="artemisApp.programmingExercise.tasksAnalysis.issues">Task issues.</span>
+        </div>
+    </ng-container>
+    <ng-template #tasksOk>
+        <div id="instruction_analysis_num-of-tasks-ok" class="editable-instruction-container__status__item">
+            <fa-icon [icon]="faCheckCircle" class="text-success me-1"></fa-icon>
+            <span class="text-info" jhiTranslate="artemisApp.programmingExercise.tasksAnalysis.ok">Tasks ok.</span>
+        </div>
+    </ng-template>
+    <ng-template #tasksTooltipContent>
+        <div class="text-start w-100">
+            <div *ngIf="numOfTasks > 15" class="text-start">
+                <span>{{ 'artemisApp.programmingExercise.tasksAnalysis.numOfTasks15' | artemisTranslate : { number: numOfTasks } }}</span>
+            </div>
+        </div>
+    </ng-template>
```

Comment:
Could this be its own small component (e.g. `task-count-warning.component`) that only gets the number of tasks as `@Input`?

Then the logic when a warning should be shown and when not is encapsulated in there and can be tested on its own in a test that checks if certain elements are present/absent in the HTML (there should be examples for similar tests in various places in the Artemis client tests).

The limit of `15` could then also be a constant in this component that is passed as additional parameter to the translation to make it more flexible for future changes.

Reply:
@b-fein I also implemented this request

Reply:
Thanks for the feedback @b-fein. This is implemented too :)